fdbserver/workloads/WriteTagThrottling.actor.cpp

```diff
@@ -0,0 +1,262 @@
+		badActorReadLatency.reset(new ContinuousSample<double>(sampleSize));
+		goodActorReadLatency.reset(new ContinuousSample<double>(sampleSize));
+		badActorCommitLatency.reset(new ContinuousSample<double>(sampleSize));
+		goodActorCommitLatency.reset(new ContinuousSample<double>(sampleSize));
+
+		writeThrottle = getOption(options, LiteralStringRef("writeThrottle"), false);
+		badActorPerClient = getOption(options, LiteralStringRef("badActorPerClient"), 1);
+		goodActorPerClient = getOption(options, LiteralStringRef("goodActorPerClient"), 1);
+		actorCount = goodActorPerClient + badActorPerClient;
+
+		keyCount = getOption(options, LiteralStringRef("keyCount"), 3000);
+		txBackoffDelay = actorCount * 1.0 / getOption(options, LiteralStringRef("txPerSecond"), 1000);
+
+		badReadTag = TransactionTag(std::string("badReadTag"));
+		badWriteTag = TransactionTag(std::string("badWriteTag"));
+		goodTag = TransactionTag(std::string("goodTag"));
+	}
+
+	virtual std::string description() { return WriteTagThrottlingWorkload::NAME; }
+
+	// choose a tag
+	ACTOR static Future<Void> _setup(Database cx, WriteTagThrottlingWorkload* self) {
+		state TransactionPriority priority = deterministicRandom()->randomChoice(allTransactionPriorities);
+		state double rate = deterministicRandom()->random01() * 20;
+		state TagSet tagSet1;
+		state TagSet tagSet2;
+		state TagSet tagSet3;
+		tagSet1.addTag(self->badWriteTag);
+		tagSet2.addTag(self->badReadTag);
+		tagSet3.addTag(self->goodTag);
+
+		wait(ThrottleApi::throttleTags(cx, tagSet1, rate, self->testDuration + 120, TagThrottleType::MANUAL, priority));
+		wait(ThrottleApi::throttleTags(cx, tagSet2, rate, self->testDuration + 120, TagThrottleType::MANUAL, priority));
+		wait(ThrottleApi::throttleTags(cx, tagSet3, rate, self->testDuration + 120, TagThrottleType::MANUAL, priority));
+
+		return Void();
+	}
+	Future<Void> setup(const Database& cx) override { return _setup(cx, this); }
+	ACTOR static Future<Void> _start(Database cx, WriteTagThrottlingWorkload* self) {
+		state vector<Future<Void>> clientActors;
+		state int actorId;
+		for (actorId = 0; actorId < self->goodActorPerClient; ++actorId) {
+			clientActors.push_back(clientActor(false, actorId, 0, cx, self));
+		}
+		for (actorId = 0; actorId < self->badActorPerClient; ++actorId) {
+			clientActors.push_back(clientActor(true, actorId, self->badOpRate, cx, self));
+		}
+		wait(delay(self->testDuration));
+		return Void();
+	}
+	virtual Future<Void> start(Database const& cx) { return _start(cx, this); }
+	virtual Future<bool> check(Database const& cx) {
+		if (badActorTxNum == 0 && goodActorTxNum == 0) {
+			TraceEvent(SevError, "NoTransactionsCommitted");
+			return false;
+		}
+		// TODO check whether write throttling work correctly after enabling that feature
+		return true;
+	}
+	virtual void getMetrics(vector<PerfMetric>& m) {
+		m.push_back(PerfMetric("Transactions (badActor)", badActorTxNum, true));
```

Comment:
For these `PerfMetric`s, the last boolean field determines whether or not the `PerfMetric` will be averaged across test clients when computing aggregate metrics. This field should not be averaged, but the latency metrics should be averaged.